src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3268,12 +3268,9 @@ def _call_handler(self, node, parent):
         try:
             routine_symbol = symbol_table.lookup(call_name)
             if type(routine_symbol) is Symbol:
-                # TODO PR #1063: Symbol should be specialised to a
-                # RoutineSymbol here (if the symbol is part of a use
-                # statement). Without specialising, the Call class
-                # constructor will raise an exception. As a temporary
-                # fix, just change the class name.
-                routine_symbol.__class__ = RoutineSymbol
+                # Specialise routine_symbol from a Symbol to a
```

Comment:
In this case if the lookup wrongly returns a DataSymbol, it will not break here and I think it should. Also there is a pylint issue. Maybe change to:
```
if not isinstance(routine_symbol, RoutineSymbol):
    routine_symbol.specialise(RoutineSymbol)
```

Reply:
Yes, good point. I've updated to check for invalid types and raise an exception if so.

Reply:
Also added a pylint disable-warning directive as I think that `type` is the right thing to use here.